client/src/js/reducers/DocumentsReducer.js

```diff
@@ -10,6 +10,15 @@ export default (state = [], action) => {
       return [...state, action.payload];
     case types.DELETE_DOCUMENT:
       return state.filter(doc => action.id !== doc.id);
+    case types.UPDATE_DOCUMENT:
+      console.log(state, 'new update state');
+      console.log(action.updatedDocument, 'new real state');
+      return state.map((doc) => {
+        if (doc.id === action.updatedDocument.id) {
```

Comment:
you can use the ternary operator to make this shorter e.g
`
return state.map((doc) =>
    (doc.id === action.updatedDocument.id) ? action.updatedDocument : doc`
)